server/src/operators/organization_operator.rs

```diff
@@ -622,3 +622,47 @@ pub async fn delete_actual_organization_query(
 
     Ok(())
 }
+
+pub async fn update_all_org_dataset_server_configs_query(
+    org_id: uuid::Uuid,
+    new_config: serde_json::Value,
+    pool: web::Data<Pool>,
+) -> Result<(), ServiceError> {
+    use crate::data::schema::datasets::dsl as datasets_columns;
+
+    let new_config_json_value = serde_json::to_value(new_config).unwrap();
+
+    let mut conn = pool.get().await.unwrap();
+
+    let new_datasets: Vec<(uuid::Uuid, serde_json::Value)> = datasets_columns::datasets
+        .filter(datasets_columns::organization_id.eq(org_id))
+        .select((
+            datasets_columns::id,
+            datasets_columns::server_configuration.concat(new_config_json_value),
+        ))
+        .load::<(uuid::Uuid, serde_json::Value)>(&mut conn)
+        .await
+        .map_err(|e| {
+            log::error!(
+                "Error loading datasets in update_all_org_dataset_server_configs: {:?}",
+                e
+            );
+            ServiceError::BadRequest("Error loading datasets".to_string())
+        })?;
+
+    for (dataset_id, new_config) in new_datasets {
+        diesel::update(datasets_columns::datasets.filter(datasets_columns::id.eq(dataset_id)))
+            .set(datasets_columns::server_configuration.eq(new_config))
+            .execute(&mut conn)
+            .await
+            .map_err(|e| {
+                log::error!(
+                    "Error updating dataset in update_all_org_dataset_server_configs: {:?}",
+                    e
+                );
+                ServiceError::BadRequest("Error updating dataset".to_string())
+            })?;
+    }
```

Comment:
Want to draw attention to this section. 

I chose to do multiple small transactions instead of one big transaction because I didn't want to ever lock the whole table. 

The downside to my approach is there it incurs overhead from so many commits for orgs with lots of datasets, but I still think it's worth avoiding the big lock. 

Other consideration is that this in theory possible with one dataset call which I want to try before this gets merged. Will be attempting here shortly. 

I think postgres supports the concat operator on a update even tho diesel does not.